sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -68,13 +82,14 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
       // the value of spark.sql.warehouse.dir.
       // When neither spark.sql.warehouse.dir nor hive.metastore.warehouse.dir is set,
       // we will set hive.metastore.warehouse.dir to the default value of spark.sql.warehouse.dir.
-      val sparkWarehouseDir = sparkContext.conf.get(WAREHOUSE_PATH)
+      val sparkWarehouseDir = conf.get(WAREHOUSE_PATH)
       logInfo(s"Setting hive.metastore.warehouse.dir ('$hiveWarehouseDir') to the value of " +
         s"${WAREHOUSE_PATH.key} ('$sparkWarehouseDir').")
-      sparkContext.hadoopConfiguration.set("hive.metastore.warehouse.dir", sparkWarehouseDir)
+      hadoopConf.set("hive.metastore.warehouse.dir", sparkWarehouseDir)
       sparkWarehouseDir
     }
   }
+  sparkContext.conf.set(WAREHOUSE_PATH.key, warehousePath)
   logInfo(s"Warehouse path is '$warehousePath'.")
 
 
```

Comment:
What is the reason we keep the line 107 `new SQLAppStatusListener(sparkContext.conf` unchanged?